React/Modules/RCTDevSettings.h

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <React/RCTBridge.h>
+
+/**
+ * This notification is posted when user settings change.
+ * It will be scoped to the RCTDevSettingsDataSource that posted it.
+ * Its userInfo dictionary may optionally contain `kRCTDevSettingsUpdatedSettingsKey`
+ * with an array of specific settings keys that changed.
+ */
+FOUNDATION_EXPORT NSNotificationName const kRCTDevSettingsDidUpdateNotification;
+FOUNDATION_EXPORT NSString * const kRCTDevSettingsUpdatedSettingsKey;
+
+@protocol RCTDevSettingsDataSource <NSObject>
+
+- (void)updateStoredSettingWithValue:(id)value forKey:(NSString *)key;
```

Comment:
nit: remove stored from the interface name. Whether a data source decides to store settings or not is an implementation choice.

Please also document these two methods, explaining when they're called / what the expected behaviour is.